ui/components/app/confirm-page-container/flask/snap-insight.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import classnames from 'classnames';
+
+import Preloader from '../../../ui/icon/preloader/preloader-icon.component';
+import Typography from '../../../ui/typography/typography';
+import { COLORS } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { useTransactionInsightSnap } from '../../../../hooks/flask/useTransactionInsightSnap';
+import SnapContentFooter from '../../flask/snap-content-footer/snap-content-footer';
+
+export const SnapInsight = ({ transaction, chainId, selectedSnap }) => {
+  const t = useI18nContext();
+  const response = useTransactionInsightSnap({
+    transaction,
+    chainId,
+    snapId: selectedSnap.id,
+  });
+
+  const data = response?.insights;
+
+  return (
+    <div
+      className={classnames('snap-insight', {
+        'snap-insight--no-data': !data || !Object.keys(data).length,
```

Comment:
`!(data && Object.keys(data).length)` is easier to understand.

Reply:
This has changed in 0bfba4c